etc/errors/anvil.json

```diff
@@ -19,7 +19,215 @@
+                  "name": "port_fallback_retried",
+                  "type": "uint"
+                },
+                {
+                  "name": "details",
+                  "type": "string"
+                }
+              ],
+              "doc": {
+                "summary": "Failed to start the server and bind it to the requested host and port.",
+                "description": [
+                  "Anvil-zksync starts the server and listens to requests on a specified host and port, 0.0.0.0:8011 by default. They are configurable using `--host` and `--port` command line arguments.",
+                  "",
+                  "The host and port used by anvil-zksync are also displayed when you start anvil-zksync:",
+                  "",
+                  "```",
+                  "========================================",
+                  "Listening on 0.0.0.0:8011",
+                  "========================================",
+                  "```",
+                  "",
+                  "This error indicates that listening on the specified host and port There are missing or invalid command line arguments, or an invalid combination of arguments is provided."
+                ],
+                "likely_causes": [
+                  {
+                    "cause": "You don't have permission to use specified host and port",
+                    "fixes": [
+                      "Check your firewall settings",
+                      "Try with a different port"
+                    ]
+                  },
+                  {
+                    "cause": "The selected port is in use",
+                    "fixes": [
+                      "Check currently running applications that might use the port you want",
+                      "Try with a different port"
+                    ]
+                  }
+                ]
+              }
+            },
+            {
+              "name": "TelemetryCollectionFailed",
+              "code": 3,
+              "message": "Telemetry collection error: {details}",
+              "fields": [
+                {
+                  "name": "details",
+                  "type": "string"
+                },
+                {
+                  "name": "inner",
+                  "type": "string"
+                }
+              ],
+              "doc": {
+                "summary": "Error collecting telemetry data.",
+                "description": "Anvil-zksync collects telemetry data to help the anvil-zksync developers improving the user experience (only if you have explicitly agreed on it). This error indicates that something went wrong when initializing telemetry subsystem or sending telemetry data.",
+                "likely_causes": [
+                  {
+                    "cause": "Not enough space in the storage.",
+                    "fixes": [
+                      "Free up some space in your storage",
+                      "Turn off sending telemetry data"
```

Comment:
Is there an easy way to do so? If the user is supposed to delete the config file, let's at least point to where that config is stored

Reply:
TBH I don't think people should receive any errors related to the telemetry at all. It's us who needs the telemetry, so any problem there is our problem, not users. If initializing/sending telemetry fails, user should not know about it. Moreover we should never ask the user who agreed to send telemetry to disable it.

Reply:
Answered in DM.

Reply:
Agreed